moto/batch/models.py

```diff
@@ -1585,7 +1585,14 @@ def list_jobs(self, job_queue, job_status=None):
         return jobs
 
     def cancel_job(self, job_id, reason):
+        if job_id is None:
+            raise ClientException("Job ID does not exist")
+        if reason is None:
+            raise ClientException("Reason does not exist")
+
         job = self.get_job_by_id(job_id)
+        if job is None:
+            raise ClientException("Job not found")
```

Comment:
I've just tested this against the AWS CLI with a random job-id, and it looks like it just returns an empty `200 OK` response. 
So it doesn't actually throw an error, it just always succeeds.

Copying that behaviour would still be a lot better than our current `AttributeError`. Would you mind adding a test case for this as well? The existing tests are in `tests/test_batch/test_jobs.py`, you can just add a small test that calls cancel_job with a random ID and assert that the response has a HTTP status code of 200.

Reply:
No problem. I have added tests to accommodate the changes I have made to `cancel_job` and `terminate_job`.